### ISSUE TITLE
v6.e.PREVIEW role tweak submethods don't work across modules

### ISSUE THREAD

**Issue by barcharcraz** [rakudo/rakudo#3365]
## The Problem
consider:
```perl6
use v6.e.PREVIEW;
unit module testmod;

our role ContainedThing is export {
    has Str $.something;
    has Str $.name;

}

our role ContainerThing is export {
    has str $.cname;
    has ContainedThing @.things;
    submethod TWEAK(Str :$name, *%rest) {
        @!things.append(ContainedThing.new(:$name, |%rest));
    }
}

our class TestThing is export {
    also does ContainerThing;
}
```

```perl6
use v6.e.PREVIEW;

use testmod;

my $t = TestThing.new(cname => "the cname", name => "the name", something => "the something");
say $t.things[0];
```
## Expected Behavior
```
ContainedThing.new(something => "the something", name => "the name")
```
## Actual Behavior
```
P6opaque: no such attribute '@!things' on type $?CLASS in a testmod::TestThing when trying to get a value
  in submethod TWEAK at C:\Users\chbarto\source\repos\curator\curator.p6\lib\testmod.pm6 (testmod) line 15
  in block <unit> at .\bin\main line 6
```

Note that if all the roles are in the same file as the testing code things work great. It's only across modules that the error happens.

## Environment
* Operating system:  Windows 10
* Compiler version (`perl6 -v`): 
```
This is Rakudo version 2019.11-277-gc5fa372de built on MoarVM version 2019.11-93-g7a93b2897
implementing Perl 6.d.
```

**Comment by vrurg** [comment]
Golfed down to:

_testmod.rakumod_:
```perl6
use v6.e.PREVIEW;
unit module testmod;

role R[::T] is export {
    submethod TWEAK {
        note "T:", T.^name;
    }
}
class C does R[Int] is export {
}

```

_test.raku_:
```perl6
use v6.e.PREVIEW;
use testmod;
C.new;
```

The problem is about `$?CLASS` and `::?CLASS` are type captures which are lost by `6.e`.

**Comment by Altai-man** [comment]
@vrurg is it a blocker?

**Comment by vrurg** [comment]
@Altai-man I would rather say "no" because `6.e` is in experimental stage. Neither `6.c` nor `6.d` are broken.

**Comment by vrurg** [comment]
Fixed with #3366
